bluej/src/main/java/bluej/compiler/Job.java

```diff
@@ -37,7 +38,7 @@ of the License, or (at your option) any later version.
  *
  * @author  Michael Cahill
  */
-record Job(CompileInputFile[] sources, Compiler compiler, CompileObserver observer, BPClassLoader bpClassLoader, File destDir,
+record Job(List<CompileInputFile> sources, Compiler javaCompiler, Compiler kotlinCompiler, CompileObserver observer, BPClassLoader bpClassLoader, File destDir,
```

Comment:
I think worth a comment here that kotlinCompiler is non-null if and only if there are Kotlin files in the job.